src/app/components/Button.js

```diff
@@ -14,9 +14,9 @@ export const Button = (
   }
   btnDomObj.innerText = btnObj.btnText;
   btnDomObj.classList.add('button');
-  btnDomObj.classList.add(...btnObj.classList);
+  if (btnObj.classList[0]) btnDomObj.classList.add(...btnObj.classList);
   btnDomObj.onclick = btnObj.onClickFn;
-
+  if (btnObj.isSpecial) btnDomObj.classList.add('button--special');
```

Comment:
To co przy tym rozwiazaniu powinno sie zrobic, to dobrze dodac wartosc isSpecial do obiektu z parametrami funkcji. No i oczywiscie napisac test :) Też więcej przypadków testowych Special / nie special sie powtarza a tak to mamy mniej.